subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/moduleconverter/dependencies/ExternalModuleIvyDependencyDescriptorFactory.java

```diff
@@ -40,7 +40,8 @@ public DslOriginDependencyMetadata createDependencyDescriptor(String clientConfi
         boolean changing = externalModuleDependency.isChanging();
         boolean transitive = externalModuleDependency.isTransitive();
 
-        ModuleVersionSelector requested = new DefaultModuleVersionSelector(nullToEmpty(dependency.getGroup()), nullToEmpty(dependency.getName()), nullToEmpty(dependency.getVersion()));
+        DefaultModuleVersionSelector requested = new DefaultModuleVersionSelector(nullToEmpty(dependency.getGroup()), nullToEmpty(dependency.getName()), nullToEmpty(dependency.getVersion()));
+        requested.setVersionConstraint(((VersionConstraintInternal)externalModuleDependency.getVersionConstraint()).normalize());
```

Comment:
This doesn't look right: why not just construct the `ModuleVersionSelector` directly with the `VersionConstraint`?